src/ngsa/app/DataAccessLayer/InMemoryDal.cs

```diff
@@ -151,120 +196,144 @@ public Task<IEnumerable<Actor>> GetActorsAsync(string q, int offset = 0, int lim
             return Task<IEnumerable<Actor>>.Factory.StartNew(() => { return res; });
         }
 
+        public Task<List<string>> GetFeaturedMovieListAsync()
+        {
+            return Task<List<string>>.Factory.StartNew(() =>
+            {
+                return new List<string> { "tt0133093", "tt0120737", "tt0167260", "tt0167261", "tt0372784", "tt0172495", "tt0317705" };
+            });
+        }
+
+        public Task<IEnumerable<string>> GetGenresAsync()
+        {
+            return Task<IEnumerable<string>>.Factory.StartNew(() => { return Genres; });
+        }
+
+        public async Task<Movie> GetMovieAsync(string movieId)
+        {
+            return await Task.Run(() =>
+            {
+                if (MoviesIndex.ContainsKey(movieId))
+                {
+                    return MoviesIndex[movieId];
+                }
+
+                throw new CosmosException("Not Found", System.Net.HttpStatusCode.NotFound, 404, string.Empty, 0);
+            }).ConfigureAwait(false);
+        }
+
+        public Task<IEnumerable<Movie>> GetMoviesAsync(MovieQueryParameters movieQueryParameters)
+        {
+            if (movieQueryParameters == null)
+            {
+                return GetMoviesAsync(string.Empty, string.Empty, offset: 0, limit: 100);
+            }
+
+            return GetMoviesAsync(movieQueryParameters.Q, movieQueryParameters.Genre, movieQueryParameters.Year, movieQueryParameters.Rating, movieQueryParameters.ActorId, movieQueryParameters.GetOffset(), movieQueryParameters.PageSize);
+        }
+
         public Task<IEnumerable<Movie>> GetMoviesAsync(string q, string genre, int year = 0, double rating = 0.0, string actorId = "", int offset = 0, int limit = 100)
         {
             List<Movie> res = new List<Movie>();
             int skip = 0;
             bool add = false;
 
-            foreach (Movie m in Movies)
+            if ((year > 0 || !string.IsNullOrEmpty(genre)) &&
+                !(year > 0 && !string.IsNullOrEmpty(genre)) &&
+                string.IsNullOrEmpty(q) &&
+                rating == 0 &&
+                string.IsNullOrEmpty(actorId) &&
+                offset == 0)
```

Comment:
Do we want the string tests to be Null or Whitespace since it also includes Empty?

Reply:
I added a todo - i've seen weird behavior with null or whitespace in a container, so will need to test.